### ISSUE TITLE
`--files-from` does not list the files with leading/trailing whitespace in its name

### ISSUE THREAD

**Issue by geart891** [rclone/rclone#3762]
#### What is the problem you are having with rclone?
`--files-from` does not list the files with leading/trailing whitespace in its name. Since `--files-from` does not support **wildcards**, there is noway to list the files with the tag.

#### What is your rclone version (output from `rclone version`)
Version tested:
1.49.5
1.50.1
1.50.2

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
Os tested:
Ubuntu x64
Windows 10 x64

####  Which cloud storage system are you using? (eg Google Drive)
Google Drive

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
`rclone lsf remote: --files-only --files-from list.txt`

`list.txt`
```
# Filenames with leading, trailing whitespace or both
 leading-whitespace-filename.txt
trailing-whitespace-filename.txt 
 leading-trailing-whitespace.txt 
```

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)
```
$ rclone lsf remote: --files-only --files-from list.txt -vv                                                        2019/11/25 05:40:40 DEBUG : rclone: Version "v1.50.2" starting with parameters ["F:\\System\\rClone\\rclone.exe" "lsf" "remote:" "--files-only" "--files-from" "list.txt" "-vv"]
2019/11/25 05:40:40 DEBUG : Using config file from "C:\\Users\\Rhaps\\.config\\rclone\\rclone.conf"
2019/11/25 05:40:42 DEBUG :  leading-trailing-whitespace.txt : Excluded
2019/11/25 05:40:42 DEBUG :  leading-whitespace-filename.txt: Excluded
2019/11/25 05:40:42 DEBUG : trailing-whitespace-filename.txt : Excluded
2019/11/25 05:40:42 DEBUG : 5 go routines active
2019/11/25 05:40:42 DEBUG : rclone: Version "v1.50.2" finishing with parameters ["F:\\System\\rClone\\rclone.exe" "lsf" "remote:" "--files-only" "--files-from" "list.txt" "-vv"]
```

P/S: A small visual bug. 
It seems that the `Elapsed time:` line does not include an eol character.
Os: Windows 10 x64, git-bash MINGW64 Shell, lastest rclone.

![bugreport](https://user-images.githubusercontent.com/27849964/69502992-c1fe8900-0f47-11ea-81fa-d1c366e1d2fa.png)

**Comment by ncw** [comment]
> `--files-from` does not list the files with leading/trailing whitespace in its name. Since `--files-from` does not support **wildcards**, there is noway to list the files with the tag.

I see what you mean.

Currently files-from
- accepts comments
- removes whitespace from start and end of line

I think I'd probably prefer to make `--files-from` not do either of those things and just read newline delimited files.  That was always the intention and that makes it play well with `rclone lsf`.

That would be a backwards incompatible change though

If you want to use wildcards then you can use "--include-from".

**Comment by geart891** [comment]
> That would be a backwards incompatible change though
I think the change is necessary and it would be consistent with the way `--files-from` functions, which is it reads the filenames separated by newline on the condition that each line contains exactly the filename (with whitespaces, or special characters).
>If you want to use wildcards then you can use `--include-from`.
This is my current solution with this type of problem with rules like:
```
*filenames.abc*
?filenames.abc?
# etc. 
```
The process of copy/move or anything with the file quickly become a nightmare if there are multiple corrupted files in the source with slightly different names (I know I am a monkey and I only have similar intellectual friends/colleagues).

**Comment by ncw** [comment]
OK, I agree..

Here is my proposal which I [posted in the forum](https://forum.rclone.org/t/proposal-for-backwards-incompatible-change-to-files-from/13111)

Currently `--files-from` parses the input files in the same way that `--include-from` does, namely it
- ignores comments starting with `#` and `;`
- strips whitespace from start and end of strings

This makes `--files-from` less useful that it might be and was never the original intention.

I propose to make the backwards incompatible change so that `--files-from` reads every line from the file ignoring comment characters and not stripping whitespace.

This will restore the expected behaviour that `rclone lsf` is good input to `--files-from` when files have leading spaces, `;` or `#`.

The expected impact of this is minimal I think as most `--files-from` lists are auto generated.

**Comment by ncw** [comment]
The `--files-from-raw` flag has been merged to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.52

Thank you @ankur0493 for implementing this :-)
